pkg/graphql/fields.go

```diff
@@ -7,16 +7,30 @@ import (
 )
 
 type RestrictedFieldsValidator interface {
-	Validate(operation *ast.Document, restrictedFields []fields.Type) (RestrictedFieldsResult, error)
+	Validate(operation, definition *ast.Document, restrictedTypes []fields.Type) (RestrictedFieldsResult, error)
 }
 
 type fieldsValidator struct {
 }
 
-func (d fieldsValidator) Validate(operation *ast.Document, restrictedFields []fields.Type) (RestrictedFieldsResult, error) {
+func (d fieldsValidator) Validate(operation, definition *ast.Document, restrictedTypes []fields.Type) (RestrictedFieldsResult, error) {
```

Comment:
RestrictedTypesResult*

Reply:
I believe it should be Fields, as we set restrictions for fields per type

Reply:
I proposed for the sake of consistency with the parameter